flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBConnectorDeserializationSchema.java

```diff
@@ -0,0 +1,727 @@
+                "Unable to convert to float from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    private LocalDate convertInstantToLocalDate(Instant instant) {
+        return convertInstantToLocalDateTime(instant).toLocalDate();
+    }
+
+    private LocalTime convertInstantToLocalTime(Instant instant) {
+        return convertInstantToLocalDateTime(instant).toLocalTime();
+    }
+
+    private LocalDateTime convertInstantToLocalDateTime(Instant instant) {
+        return instant.atZone(TIME_ZONE).toLocalDateTime();
+    }
+
+    private Instant convertToInstant(BsonTimestamp bsonTimestamp) {
+        return Instant.ofEpochSecond(bsonTimestamp.getTime() * 1000L);
+    }
+
+    private Instant convertToInstant(BsonDateTime bsonDateTime) {
+        return Instant.ofEpochMilli(bsonDateTime.getValue());
+    }
+
+    /**
+     * A conversion from DateType to int describes the number of days since epoch.
+     *
+     * @see org.apache.flink.table.types.logical.DateType
+     */
+    private int convertToDate(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            Instant instant = convertToInstant(docObj.asDateTime());
+            return (int) convertInstantToLocalDate(instant).toEpochDay();
+        }
+        if (docObj.isTimestamp()) {
+            Instant instant = convertToInstant(docObj.asTimestamp());
+            return (int) convertInstantToLocalDate(instant).toEpochDay();
+        }
+        throw new IllegalArgumentException(
+                "Unable to convert to date from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    /**
+     * A conversion from TimeType to int describes the number of milliseconds of the day.
+     *
+     * @see org.apache.flink.table.types.logical.TimeType
+     */
+    private int convertToTime(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            Instant instant = convertToInstant(docObj.asDateTime());
+            return convertInstantToLocalTime(instant).toSecondOfDay() * 1000;
+        }
+        if (docObj.isTimestamp()) {
+            Instant instant = convertToInstant(docObj.asTimestamp());
+            return convertInstantToLocalTime(instant).toSecondOfDay() * 1000;
+        }
+        throw new IllegalArgumentException(
+                "Unable to convert to time from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    private TimestampData convertToTimestamp(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            return TimestampData.fromEpochMillis(docObj.asDateTime().getValue());
+        }
+        if (docObj.isTimestamp()) {
+            return TimestampData.fromEpochMillis(docObj.asTimestamp().getTime() * 1000L);
```

Comment:
What does the underlying timestamp value means? Is there any documentation to reference? 

Usually, `TimestampData.fromEpochMillis` is used for `TIMESTAMP_LTZ` types where the epoch milliseconds represent Java `Instant`. We recommend to use `TimestampData.fromLocalDateTime` for `TIMESTAMP` types.

Reply:
https://docs.mongodb.com/manual/reference/bson-types/

**Timestamp**

> BSON has a special timestamp type for internal MongoDB use and is not associated with the regular Date type. This internal timestamp type is a 64 bit value where:
>- the most significant 32 bits are a time_t value (seconds since the Unix epoch)
>- the least significant 32 bits are an incrementing ordinal for operations within a given second.

The BSON timestamp type is represented by a java long type. High 32 bit's represent unix epoch seconds and low 32 bits's represent an incrementing value to keep time increments and avoid duplication. 

____________

**DateTime**

>BSON Date is a 64-bit integer that represents the number of milliseconds since the Unix epoch (Jan 1, 1970). This results in a representable date range of about 290 million years into the past and future.

Reply:
Thanks Jark to for the detailed explanation of `TIMESTAMP_LTZ`.
As the documentation mentioned above, they are semantically equivalent:
- BSON `DateTime`  - `TIMESTAMP_LTZ(3)`
- BSON `Timestamp` -`TIMESTAMP_LTZ(0)`

We treated BSON `Timestamp` as TIMESTAMP_LTZ(0) because it's minimum precision is seconds and the incrementing value does not represent the exact time interval.

So, I think the transformation in the code can work. 
If there is a deviation in my understanding, please help to correct me.

Reply:
Thanks @Jiabao-Sun , I think it makes sense to map BSON `DateTime` and `Timestamp` to Flink `TIMESTAMP_LTZ` because they all represents the seconds since epoch. However, if we map them to Flink `TIMESTAMP` (i.e. TIMESTAMP WITHOUT TIME ZONE), then we need to do a conversion (epoch seconds to local clock time) just like Java `Instant` to Java `LocalDateTime`. This conversion requires a time zone which means what time zone to dispaly the instant in string. Usually, the time zone is configured by users and not hard code, e.g. [table.local-time-zone](https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/table/config/#table-local-time-zone) in Flink SQL , and [server-time-zone](https://ververica.github.io/flink-cdc-connectors/release-2.0/content/connectors/mysql-cdc.html) in mysql-cdc.

Reply:
Thanks @wuchong, I'll add a `server-time-zone` configuration for users to configure.

Reply:
https://docs.mongodb.com/manual/reference/bson-types/#timestamps

Reply:
Thanks @wuchong for the detailed explanation.

I added a new config option `local-time-zone` to let users decide which time zone to convert to `TIMESTAMP`.

I think this is semantically more similar to `table.local-time-zone` in Flink SQL rather than `server-time-zone` of MySQL, so I named it as `local-time-zone`.

By the way, is there a method provided by Flink to get the `table.local-time-zone` at runtime, so we may not need to add additional config option.

Reply:
Yes, you can get it by `org.apache.flink.table.factories.DynamicTableFactory.Context#getConfiguration`.

Reply:
Thanks @wuchong.

Removed table option 'local-time-zone', use time zone of `table.local-time-zone` at runtime.